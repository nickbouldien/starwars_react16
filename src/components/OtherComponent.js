import React from 'react';
import { Image, Segment, Header, Divider } from 'semantic-ui-react';
import RenderText from './RenderText';

const OtherComponent = ({ title, imgSource }) => (
  <Segment>
    <Header as="h3">{title}</Header>
    <Image src={imgSource} alt={"star wars gif"} />
    <Divider />
    <RenderText straightText="yessir" />
  </Segment>
);

export default OtherComponent;
