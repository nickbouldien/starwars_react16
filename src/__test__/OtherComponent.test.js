// Frag.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import { mount, render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import setup from './setup';

import OtherComponent from '../components/OtherComponent';

const PROPS = {
  title: "Cool title",
  imgSource: "./img/cool_pic.png" ,
}

test('sanity check', () => {
  expect(true).toBe(true);
});

describe('<OtherComponent /> component', () => {
  it('renders (shallow) the OtherComponent component', () => {
    const wrapper = shallow(<OtherComponent {...PROPS} />);
    // console.log('wrapper', toJson(wrapper.children()));
    expect(wrapper.children().length).toBe(4);
  });
// });

test('OtherComponent matches snapshot', () => {
  const component = mount(<OtherComponent {...PROPS} /> );

  // console.log('h3: ', component.find('h3').text());
  expect(component.find('h3').text()).toBe('Cool title');
  expect(component.find('img').props().src).toEqual('./img/cool_pic.png');

  expect(toJson(component)).toMatchSnapshot();
});

//   let tree = component.toJSON();
//   expect(tree).toMatchSnapshot();

//   // manually trigger the callback
//   // tree.props.onMouseEnter();
//   // // re-rendering
//   // tree = component.toJSON();
//   // expect(tree).toMatchSnapshot();

//   // // manually trigger the callback
//   // tree.props.onMouseLeave();
//   // // re-rendering
//   // tree = component.toJSON();
//   // expect(tree).toMatchSnapshot();
  
});