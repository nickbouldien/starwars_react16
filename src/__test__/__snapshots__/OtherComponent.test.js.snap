// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OtherComponent /> component OtherComponent matches snapshot 1`] = `
<OtherComponent
  imgSource="./img/cool_pic.png"
  title="Cool title"
>
  <Segment>
    <div
      className="ui segment"
    >
      <Header
        as="h3"
      >
        <h3
          className="ui header"
        >
          Cool title
        </h3>
      </Header>
      <Image
        alt="star wars gif"
        as="img"
        src="./img/cool_pic.png"
        ui={true}
      >
        <img
          alt="star wars gif"
          className="ui image"
          src="./img/cool_pic.png"
        />
      </Image>
      <Divider>
        <div
          className="ui divider"
        />
      </Divider>
      <RenderText
        straightText="yessir"
      >
        Not even the younglings survived (a component)
      </RenderText>
    </div>
  </Segment>
</OtherComponent>
`;
